### PR TITLE
Release 2.13.3

### PR DESCRIPTION
**Pull request by julianbrost**
Changelog and version bump.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
 
 Released closed milestones can be found on [GitHub](https://github.com/Icinga/icinga2/milestones?state=closed).
 
+## 2.13.3 (2022-04-14)
+
+This version includes bugfixes for many features of Icinga 2, including fixes for multiple crashes. It also includes a
+number of fixes and improvements for Icinga DB.
```
Comment (Al2Klimov) [inline CHANGELOG.md:13]: I'd split this line(s) by `.`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+
+### API
+
+* The /v1/config/stages endpoint now immediately rejects parallel config updates instead of accepting and then later
+  failing to verify and activate them. #9328
```
Comment (Al2Klimov) [inline CHANGELOG.md:18]: Please rethink your strategy as in https://github.com/Icinga/icinga2/pull/8944#discussion_r681002942 .

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+
+* The lifetime of newly issued node certificates is reduced from 15 years to 397 days. #9337
+* Compare cluster certificate tickets in constant time. #9333
+
+### Notifications
+
+* Fix a crash that could happen while sending notifications shortly after Icinga 2 started. #9124
+* State notifications are sent more reliably now after notifications are resumed, for example when a downtime ends.
+  This fixes missing or redundant notifications after certain combinations of state changes happened while notifications
+  were suppressed. #9285
```
Comment (Al2Klimov) [inline CHANGELOG.md:30]: Isn't the second sentence enough?

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+
+### Downtimes
+
+* Fix scheduling of downtimes for all services on child hosts. #9159
+* Fix an issue when creating fixed downtimes starting immediately that resulted in a missing notification. #9158
+
+### Time Periods
+
+* Fix some issues involving daylight saving time changes that could result in an hour missing from the time period.
+  This fix applies to downtimes as well. #9238
```
Comment (Al2Klimov) [inline CHANGELOG.md:46]: **Scheduled** downtimes.

Comment (Al2Klimov) [inline CHANGELOG.md:46]: Also consider flipping the _as well_ by putting downtimes first...

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+### Checks and Commands
+
+* Fix a deadlock when processing check results for checkables with dependencies. #9228
+* Fix a message routing loop that can happen for event commands that are executed within a zone using `command_endpoint`
+  that resulted in excessive execution of the command. #9260
+
+### Downtimes
+
+* Fix scheduling of downtimes for all services on child hosts. #9159
+* Fix an issue when creating fixed downtimes starting immediately that resulted in a missing notification. #9158
```
Comment (Al2Klimov) [inline CHANGELOG.md:41]: ... to merge with this section.

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+
+### API
+
+* The /v1/config/stages endpoint now immediately rejects parallel config updates instead of accepting and then later
+  failing to verify and activate them. #9328
+
+### Certificates
+
+* The lifetime of newly issued node certificates is reduced from 15 years to 397 days. #9337
+* Compare cluster certificate tickets in constant time. #9333
```
Comment (Al2Klimov) [inline CHANGELOG.md:23]: Apropos merges: cluster, ...

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+* Fix an issue when building within an unrelated Git repository, version information from that repository could
+  incorrectly be used for Icinga 2. #9155
+* Windows: Update bundled Boost version to 1.78.0 and OpenSSL to 1.1.1n #9325
+
+### Internals
+
+* Fix some race conditions due to missing synchronization. Note: these race conditions should not have caused any
+  practical problems besides incorrect numbers in debug log message. #9306
+* Move the startup.log and status files created when validating incoming cluster config updates to /var/lib/icinga2/api
+  and always keep the last failed startup.log to ease debugging. #9335
```
Comment (Al2Klimov) [inline CHANGELOG.md:80]: ... internals and maybe a misc section.

File: CHANGELOG.md [hunk 8]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+* Fix the evaluation order of default templates when used in combination with apply rules. Now default templates are
+  imported first as stated in the documentation and as it already happens for objects defined without using apply. #9290
+
+### IDO
+
+* Fix an issue where contacts were not written correctly to the notification history if multiple IDO instances are
+  active on the same node. #9242
+* Explicitly set the encoding for MySQL connections as a workaround for changed defaults in Debian bullseye. #9312
+* Previous versions shipped MySQL schema files with inconsistent version information in the full schema file and upgrade
+  files. This could result in inaccurate reports of an outdated schema version. #9139
```
Comment (Al2Klimov) [inline CHANGELOG.md:59]: Doesn’t everything else focus on the present, not the past?

File: CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+
+### Performance Data Writers
+
+* Fix a race condition in the InfluxDB Writers that could result in a crash. #9237
+* Fix a log message where Influxdb2Writer logged as InfluxdbWriter. #9315
+* Fix an issue in all writers that could result in metrics being written multiple times in non-default HA setups. #9322
```
Comment (Al2Klimov) [inline CHANGELOG.md:65]: IMAO there aren’t default and non-default HA setups, but only HA setups.

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+  practical problems besides incorrect numbers in debug log message. #9306
+* Move the startup.log and status files created when validating incoming cluster config updates to /var/lib/icinga2/api
+  and always keep the last failed startup.log to ease debugging. #9335
+
+### Icinga DB
+
+* The severity attribute was updated to match the sort order Icinga Web 2 uses for the IDO. The documentation for this
+  attribute was already incorrect before and was updated to reflect the current functionality. #9239 #9240
+* Fix the `is_sticky` attribute for comments. #9303
+* Fix missing updates of is_reachable and severity in the state tables. #9241
```
Comment (Al2Klimov) [inline CHANGELOG.md:87]: `To quote` or not to quote... this is the question.

File: CHANGELOG.md [hunk 11]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+* Fix an issue where removing an acknowledgement would write incorrect comment history. #9302
+* Fix multiple issues so that in a HA zone, both nodes now write consistent history. #9157 #9182 #9190
+* Fix an issue where instead of updating the state information, a history event was written. #9252
+* Fix an issue where incomplete comment history events were generated. Note: when removing comments using the API, the
+  dedicated remove-comment action should be used instead of the objects API, otherwise no history event for that action
+  will be generated. #9301
+* Fix an issue where the order attribute of command arguments could not be handled if it wasn't an integer. Note: You
+  should only specify integer values for order, other values are converted to integer before use so using fractional
+  numbers there has no effect. #9181
```
Comment (Al2Klimov) [inline CHANGELOG.md:96]: Consider shortening out _an issue where_, after all all bullet points here are about issues.

File: CHANGELOG.md [hunk 12]
```diff
@@ -7,6 +7,98 @@ documentation before upgrading to a new release.
...
+* The severity attribute was updated to match the sort order Icinga Web 2 uses for the IDO. The documentation for this
+  attribute was already incorrect before and was updated to reflect the current functionality. #9239 #9240
+* Fix the `is_sticky` attribute for comments. #9303
+* Fix missing updates of is_reachable and severity in the state tables. #9241
+* Fix an issue where removing an acknowledgement would write incorrect comment history. #9302
+* Fix multiple issues so that in a HA zone, both nodes now write consistent history. #9157 #9182 #9190
+* Fix an issue where instead of updating the state information, a history event was written. #9252
+* Fix an issue where incomplete comment history events were generated. Note: when removing comments using the API, the
+  dedicated remove-comment action should be used instead of the objects API, otherwise no history event for that action
+  will be generated. #9301
```
Comment (Al2Klimov) [inline CHANGELOG.md:93]: Consider making the note bold, on a separate Markdown line (not bullet point) and after #PR_NR.

File: CHANGELOG.md (hunk 6 above)
Comment (julianbrost) [reply CHANGELOG.md:23]: Your suggestion is to merge "API" and "Certificates" into a single "Cluster" section? If so: not sure about that, all changes (except maybe the last) can affect single node icinga2 setups as well.

Comment (Al2Klimov) [reply CHANGELOG.md:23]: Valid point.

File: CHANGELOG.md [hunk 13]
```diff
@@ -7,6 +7,101 @@ documentation before upgrading to a new release.
...
+* Ship a MySQL schema upgrade that fixes inconsistent version information in the
+  full schema file and upgrade files which could have resulted in inaccurate reports
+  of an outdated schema version. #9139
+
+### Performance Data Writers
+
+* Fix a race condition in the InfluxDB Writers that could result in a crash. #9237
+* Fix a log message where Influxdb2Writer logged as InfluxdbWriter. #9315
+* Fix an issue in all writers that could result in metrics being written multiple times
+  in HA setups if `enable_ha` is explicitly enabled on the writer. #9322
```
Comment (Al2Klimov) [inline CHANGELOG.md:65]: IMAO _in HA setups_ is enough.

File: CHANGELOG.md [hunk 14]
```diff
@@ -7,6 +7,101 @@ documentation before upgrading to a new release.
...
+
+* Fix some race conditions due to missing synchronization.
+  These race conditions should not have caused any practical problems
+  besides incorrect numbers in debug log message. #9306
+* Move the startup.log and status files created when validating incoming cluster config updates
+  to /var/lib/icinga2/api and always keep the last failed startup.log to ease debugging. #9335
+
+### Icinga DB
+
+* The severity attribute was updated to match the sort order Icinga Web 2 uses for the IDO.
```
Comment (Al2Klimov) [inline CHANGELOG.md:85]: And `this one`?

**Review by Al2Klimov — COMMENTED** [review]
Consider merging one-bullet sections to _misc_.

File: CHANGELOG.md (hunk 13 above)
Comment (Al2Klimov) [reply CHANGELOG.md:65]: I'd at least put the clarification in ( ).

**Comment by julianbrost** [comment]
> Consider merging one-bullet sections to _misc_.

I think the config packages API change is too important for burying it in a misc section. It's one of the changes you might actually observe apart from things working better: you may now see new error messages when deploying config with the Director.

**Pull request merged**